src/utils/http.ts

```diff
@@ -32,25 +35,41 @@ const convertAxiosError = (e: AxiosError): Error => {
     );
 };
 
+
+/**
+ * Normalizes the total size of a download event, and calls the provided onProgress callback with
+ * loaded and total sizes.
+ *
+ * @param onProgress
+ * @return The handler that wraps `onProgress`.
+ */
+const normalizeTotalSize = (onProgress: ProgressCallback) => ({
+    loaded,
+    total,
+}: AxiosProgressEvent) => {
+    if ("undefined" === typeof total || isNaN(total)) {
```

Comment:
Note we also added the `isNan(total)` check here. The `NaN` value was discovered through new unit tests (to be reviewed at #177 )